templates/cicd/pipeline.yml

```diff
@@ -6,18 +6,19 @@ version: {{.Version}}
 # This section defines the source artifacts.
 source:
   # The name of the provider that is used to store the source artifacts.
-  provider: {{.Provider}}
+  provider: {{.Source.ProviderName}}
   # Additional properties that further specifies the exact location
   # the artifacts should be sourced from.
   properties:
     # Repository in the format of <user_name>/<repository_name>.
-    repository: {{.Repository}}
+    repository: {{index .Source.Properties "repository"}}
```

Comment:
How would we handle fields that're not common between providers?  
I think adding `{{eq .Source.ProviderName "github"}}` is not bad could that allow us to get rid of the `fatih/structs` dependency?

Reply:
I think I might change the `properties` to be of type `map[string]interface{}` and don't explicitly write the `repository` field here.